headers/types/common/common.h

```diff
@@ -755,6 +755,50 @@ struct adventure_log {
 };
 ASSERT_SIZE(struct adventure_log, 636);
 
+struct monster_synth_data {
+    // 0x0: Number of Exclusive Items Checked
+    int32_t num_items_looped;
+    // 0x4: Sum of all num_to_trade in stored templates
+    int32_t total_num_to_trade;
+    // 0x8: synth_templates with exclusive items applicable to the species
+    struct synth_template* applicable_templates[5];
+};
+ASSERT_SIZE(struct monster_synth_data, 28);
+
+struct type_synth_data {
+    // 0x0: A bitfield of which types have species exclusive items.
+    // Presumably for determining type-specific exclusive items? Croagunk doesn't sell these...
+    bool type_is_available[20];
+};
+ASSERT_SIZE(struct type_synth_data, 20);
+
+struct baby_exclusive_item_pair {
+    struct item_id_16 exc_item_id;
+    struct monster_id_16 baby_monster;
+};
+ASSERT_SIZE(struct baby_exclusive_item_pair, 4);
+
+struct synth_template {
+    // 0x0: Item ID of the exclusive item.
+    struct item_id_16 exc_item_id;
+    // 0x2: Weight for that item to appear in the swap shop.  
+    int16_t item_weight;
+    // 0x4: Seem to be powers of 2, some kind of bitfield perhaps?
+    int16_t unk_0x4;
+    // 0x6: Minimum $SCENARIO_BALANCE_FLAG value for it to spawn.
+    // Only used for the Water Float (6) and the Phione/Manaphy items (7).
+    int16_t min_balance_flag;
+};
+ASSERT_SIZE(struct synth_template, 8);
+
+// Structure of SYNTH/synth.bin file, which is used for the Croagunk Swap Shop.
+struct synth_file {
+    struct synth_template synth_templates[780];
+    // The last 4 templates appear to be empty, with 0 weight.
+    struct synth_template unused_templates[4];
```

Comment:
There's no point in separating out the array just to denote the empty entries, considering they're all the same type. It's simpler to just make everything one big array, with a comment